content/en/docs/components/pipelines/concepts/component.md

```diff
@@ -2,66 +2,178 @@
 
-A component specification in YAML format describes the component for the
-Kubeflow Pipelines system. A component definition has the following parts:
+A KFP component consist of the following key elements:
+
+### 1. Code
+- Typically a Python function, but can be other code such as a Bash command.
+
+### 2. Dependency Support
+- **Python libraries** - to be installed at runtime
+- **Environment variables** - to be available in the runtime environment
+- **Python package indices** - (for example, private PyPi servers) if needed to support installations
+- **Cluster resources** - to support use of ConfigMaps, Secrets, PersistentVolumeClaims, and more
+- **Runtime dependencies** - to support CPU, memory, and GPU requests and limits
+
+### 3. Base Image
+- Defines the base container runtime environment (defaults to a generic Python base image)
+- May include system dependencies and pre-installed Python libraries
+
+### 4. Input/Output (I/O) Specification
+- Individual components cannot share in-memory data with each other, so they use the following concepts to support exchanging information and publishing results:
+  - **Parameters** – for small values
+  - **[Artifacts][artifacts]** - for larger data like model files, processed datasets, and metadata
+
+## Constructing a Component
+
+### 1. Python-Based Components
+The recommended way to define a component is using the `@dsl.component` decorator from the [KFP Python SDK][KFP SDK]. Below are two basic component definition examples:
+```python
+from kfp.dsl import component, Output, Dataset
+
+# hello world component
+@component()
+def hello_world(name: str = "World") -> str:
+    print(f"Hello {name}!")
+    return name
+
+# process data component
+@component(
+    base_image="python:3.12-slim-bookworm",
+    packages_to_install=["pandas>=2.2.3"],
+)
+def process_data(output_data: Output[Dataset]):
+    '''Get max from an array'''
+    import pandas as pd
+   # create dataset to write to output
+    data = pd.DataFrame(data=[[1,2,3],[4,5,6]], columns=["a","b","c"])
+    data.to_csv(output_data.path)
+```
+
+Observe that these are wrapped Python functions. The `@component` wrapper helps the KFP Python SDK supply the needed context for running these functions in containers as part of a KFP [pipeline][pipeline].
+
+The `hello_world` component just uses the default behavior, which is to run the Python function on the default base image (`python:3.9`). 
+
+The `process_data` component adds layers of customization, by supplying the name of a specific `base_image`, and `packages_to_install`. Note the inclusion of the `import pandas as pd` statement inside the function; since the function will run inside a container (and won't have the script context), all non-builtin Python library dependencies need to be imported within the component function. This component also uses KFP's `Output[Dataset]` class, which takes care of creating a KFP [artifact][artifacts] type output. 
```

Comment:
It was confusing me since you still need to import the Python standard library.
```suggestion
The `process_data` component adds layers of customization, by supplying the name of a specific `base_image`, and `packages_to_install`. Note the inclusion of the `import pandas as pd` statement inside the function; since the function will run inside a container (and won't have the script context), all Python library dependencies need to be imported within the component function. This component also uses KFP's `Output[Dataset]` class, which takes care of creating a KFP [artifact][artifacts] type output. 
```